### ISSUE TITLE
Incorrect number of KSAnnotation#arguments for kotlin.Metadata

### ISSUE THREAD

**Issue by wanyingd1996** [google/ksp#1754]
Hi, all. I annotated a class with kotlin.Metadata, and was using ksp to process the annotation, but found the corresponding KSAnnotation#arguments is returning incorrect results.

I used `ksAnnotated.arguments.mapNotNull { it.name?.asString() }.joinToString()`, and got
```
k, kind, metadataVersion, bytecodeVersion, data1, data2, extraString, packageName, extraInt
```

There are 9 arguments returned, but it is supposed to have 8 arguments. Also there is a jvmName `k` got into the list.

**Comment by neetopia** [comment]
just to confirm, is this for KSP1 or KSP2?

**Comment by neetopia** [comment]
I mean did you run with `ksp.useKSP2=true` property?

**Comment by wanyingd1996** [comment]
No, we are still using Ksp1. Is this a known issue fixed with Ksp2?

**Comment by neetopia** [comment]
no, but it is a good direction for investigation. Thanks for confirming.

**Comment by wanyingd1996** [comment]
Hi, any update on this issue? This affects our processor's interoperability with other processors.
